File: tests/benchmark/vulnerable/javascript/path_traversal_001.js
// BENCHMARK: vulnerable - path_traversal
const express = require('express');
const fs = require('fs');
const app = express();

app.get('/file/:name', (req, res) => {
    const filename = req.params.name;
    fs.readFile('/var/data/' + filename, 'utf8', (err, data) => {
        if (err) return res.status(404).send('Not found');
        res.send(data);
    });
});

app.listen(3000);

File: tests/benchmark/vulnerable/go/path_traversal_001.go
// BENCHMARK: vulnerable - path_traversal
package main

import (
	"fmt"
	"net/http"
	"os"
)

func fileHandler(w http.ResponseWriter, r *http.Request) {
	filename := r.URL.Query().Get("path")
	f, err := os.Open("/var/data/" + filename)
	if err != nil {
		http.Error(w, "not found", 404)
		return
	}
	defer f.Close()
	content := make([]byte, 4096)
	n, _ := f.Read(content)
	fmt.Fprintf(w, "%s", content[:n])
}

func main() {
	http.HandleFunc("/file", fileHandler)
	http.ListenAndServe(":8080", nil)
}
